### PR TITLE
Followup fixes for python 3

### PR DESCRIPTION
**Pull request by benson-basis**
* stop using `future` except in tests, use `six` instead
* fix py2 vs py3 issues with `decode` in `setup.py`, might fix windows. Might fix android.

### CODE REVIEW

**Comment by benson-basis** [comment]
Android succeeds at create with these changes.

**Comment by dessant** [comment]
We could avoid using six in setup.py, you could declare `PY3 = sys.version_info >= (3,0,0)` somewhere at the start and use that boolean for further checks.

**Comment by dessant** [comment]
It does fail for windows even with pyhon 2, but i think that is not caused by your code, I've tried to install an older commit and it is just as broken. I'll post some error logs regarding that, but the priority is android, linux and osx for now.

File: .travis.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ before_install:
   - sudo apt-get install python-pip openjdk-7-jdk
 
 install:
-  - pip install --upgrade cython future
+  - pip install --upgrade cython future six
```
Comment (akshayaurora) [inline .travis.yml]: future not needed anymore?

**Comment by akshayaurora** [comment]
@benson-basis thanks for this,  could we remove future dependency all together? instead of using both future and six?

**Comment by benson-basis** [comment]
It's time-consuming to get all the 'future' out of the tests. I'll do it,
but I wanted to unblock things first.

**Comment by benson-basis** [comment]
@dessant I think this should solve everything mentioned except for adroid and getting rid of future from the tests. As noted elsewhere, I'm a bit stumped on the android side.

**Comment by inclement** [comment]
Thanks for the fixes. I've manually removed the apk stuff in the Makefile though as it's out of place (especially with the hardcoded machine-specific paths) - it would be better to add a test of some kind, or perhaps a pyjnius-specific test in p4a itself.

**Comment by benson-basis** [comment]
Thanks.

**Pull request merged**